feat: support JS i18n

Zacharis278 opened the pull request:
Adds support for translating rendered JavaScript. I got this approach by looking at how the https://github.com/openedx/xblock-drag-and-drop-v2 xblock does this.

Detailed Steps
- Adds a new task to make compile_translations that will generate JS translation files from the existing django .po files. Each JS file is a script that when loaded into the browser populates a window object containing all the translations. These are the files inside `public/js/translations`. They will now update alongside the python .mo files when translations are compiled.
- In order to run this command we need django app settings. Since this is just an installed library/xblock I've added a manage.py and django settings files solely for the purposes of compiling translations locally.
- Updated the student_view in lti_xblock.py to include the correct language script file when building it's html fragment.
- Updated the JS code (xblock_lti_consumer.js) to use the local translations for this library in addition to the existing edx-platform JS translations already on the window object.

Also commit the compiled (.mo) files from the last translations update. These should have gotten updated alongside the .po files.

Zacharis278 commented on .tx/config:
```diff
@@ -6,4 +6,3 @@ file_filter = lti_consumer/translations/<lang>/LC_MESSAGES/text.po
 source_file = lti_consumer/translations/en/LC_MESSAGES/text.po
 source_lang = en
 type        = PO
-
```
NIT

Zacharis278 commented on lti_consumer/static/js/xblock_lti_consumer.js:
```diff
@@ -154,7 +173,7 @@ function LtiConsumerXBlock(runtime, element) {
         function renderPIIConsentPromptIfRequired(onSuccess, showCancelButton=true) {
             if (askToSendUsername && askToSendEmail) {
-                msg = gettext('Click OK to have your username and e-mail address sent to a 3rd party application.');
+                msg = gettext('Click OK to have your uuuuuusername and e-mail address sent to a 3rd party application.');
```
fix

MichaelRoytman commented on lti_consumer/settings.py:
```diff
@@ -0,0 +1,81 @@
+BASE_DIR = os.path.dirname(os.path.dirname(__file__))
+
+
+# Quick-start development settings - unsuitable for production
```
Are you planning to "productionize" these settings?

Zacharis278 replied:
these are solely used to run translation compilation. I can make that a bit clearer in the docstring

Zacharis278 replied:
I moved this entire file under the locale folder so you don't even see it unless you're messing with translations.

MichaelRoytman commented on lti_consumer/static/js/xblock_lti_consumer.js:
```diff
@@ -1,4 +1,23 @@
 function LtiConsumerXBlock(runtime, element) {
+    var gettext;
+    console.log('hello')
```
TODO: Remove the logs.

alangsto replied:
NIT: this is all magic to me, but should these logs be left in?

MichaelRoytman commented on lti_consumer/translations/en/LC_MESSAGES/text.po:
```diff
@@ -621,13 +621,13 @@ msgstr ""
 "Click OK to have your username and e-mail address sent to a 3rd party "
 "application."
-msgstr ""
+msgstr "yesss"
```
I'm not really sure this is within the scope of your work, but these two translations look off.

alangsto replied:
were these approved?

alangsto commented on manage.py:
```diff
@@ -0,0 +1,13 @@
+        "lti_consumer.settings"
+    )
+
+    execute_from_command_line(sys.argv)
```
NIT: new line

alangsto commented on requirements/base.in:
```diff
@@ -17,3 +17,4 @@ django-filter
 jsonfield
 django-config-models         # Configuration models for Django allowing config management with auditing
 openedx-filters
+django-statici18n
```
NIT: new line

Zacharis278 commented on lti_consumer/lti_xblock.py:
```diff
@@ -645,6 +646,22 @@ def workbench_scenarios():
         return scenarios
 
+    @staticmethod
+    def _get_statici18n_js_url(loader): # pragma: no cover
```
I ended up marking this as no-cover in tests. This just got really messing trying to gain coverage on for little value since some of these functions are mocked in a test environment.

Zacharis278 commented:
@alangsto @MichaelRoytman thanks for taking an initial look at this. I went through and cleaned things up a bit and added a better description of the 'why' for all these changes.

shadinaif commented:
Many thanks for taking care of this @Zacharis278 ! Please take a look at my new PR https://github.com/openedx/xblock-lti-consumer/pull/371
